Gems/Stars/Code/Source/EditorStarsComponent.cpp

```diff
@@ -0,0 +1,114 @@
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &StarsComponentConfig::m_starsAsset, "Stars Asset", "Stars asset")
+                    ;
+
+                editContext->Class<StarsComponentController>(
+                    "StarsComponentController", "")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                        ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &StarsComponentController::m_configuration, "Configuration", "")
+                        ->Attribute(AZ::Edit::Attributes::Visibility, AZ::Edit::PropertyVisibility::ShowChildrenOnly)
+                    ;
+
+                editContext->Class<EditorStarsComponent>(
+                    "Stars", "Renders stars in the background")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "Atom")
+                    ->Attribute(AZ::Edit::Attributes::Icon, "Icons/Components/Component_Placeholder.svg")
+                    ->Attribute(AZ::Edit::Attributes::ViewportIcon, "Icons/Components/Viewport/Component_Placeholder.svg")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("Game", 0x232b318c));
+            }
+        }
+    }
+
+    void EditorStarsComponent::Init()
+    {
+        StarsComponentConfig& config = m_controller.m_configuration;
+
+        // prefill with the default stars asset if no other is specified
+        if (!config.m_starsAsset.GetId().IsValid())
+        {
+            Data::AssetId assetId;
+            const auto type = azrtti_typeid<StarsAsset>();
+            const auto path = m_defaultAssetPath.c_str();
+            Data::AssetCatalogRequestBus::BroadcastResult( assetId, &Data::AssetCatalogRequests::GetAssetIdByPath, path, type, false);
+            if (assetId.IsValid())
+            {
+                config.m_starsAsset = Data::AssetManager::Instance().FindOrCreateAsset<StarsAsset>(assetId, Data::AssetLoadBehavior::Default);
+            }
+        }
+
+        // remember the stars asset id so we can detect when it changes
+        m_prevAssetId = config.m_starsAsset.GetId();
+    }
+
+    u32 EditorStarsComponent::OnConfigurationChanged()
+    {
+        if (m_prevAssetId != m_controller.GetConfiguration().m_starsAsset.GetId())
+        {
+            m_controller.OnStarsAssetChanged();
+            m_prevAssetId = m_controller.GetConfiguration().m_starsAsset.GetId();
+        }
+
+        m_controller.OnConfigChanged();
+
+        return Edit::PropertyRefreshLevels::AttributesAndValues;
+    }
+
+    void EditorStarsComponent::OnEntityVisibilityChanged(bool visibility)
+    {
+        if (visibility)
+        {
+            m_controller.RegisterFeatureProcessor(GetEntityId());
+        }
+        else
+        {
+            m_controller.UnregisterFeatureProcessor();
+        }
+    }
```

Comment:
I was using this method originally but I think when the feature processor is registered it has all of it's activate and override methods called every frame and it was suggested in early reviews to take this approach to only register the feature processor when the component is actually used, so there is no runtime cost if it isn't used.  Eventually, I do want to better support multiple star components.

Reply:
Unfortunately, I tested this change and simply by registering any feature processor it will be activated in the default scene because in `BootstrapSystemComponent::GetOrCreateAtomSceneFromAzScene` the scene calls `EnableAllFeatureProcessors()`.  This defeats the purpose of only enabling the feature processor when the component is used, so for now I'll leave this code as is.